src/model/model.rs

```diff
@@ -182,6 +192,7 @@ impl MonitorModel {
     }
 
     pub fn update_network_status(&mut self, net_status: DeviceNetworkStatus) {
+        self.dpc_key = Some(net_status.dpc_key.clone());
```

Comment:
why clone dpc_key?

Reply:
well, the answer to "why clone()..." question in 90% of cases is "to satisfy borrow checker". But this is a really good catch! I rewrote it without cloning.